src/server/cluster/cluster_shard_migration.h

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+#pragma once
+
+#include "base/io_buf.h"
+#include "server/protocol_client.h"
+
+namespace dfly {
+
+class ClusterShardMigration : public ProtocolClient {
```

Comment:
we should have more comments in the code.
Add what this class is about and a comment on StartSyncFlow at least